spec/controllers/omniauth_callbacks_controller_spec.rb

```diff
@@ -0,0 +1,37 @@
+require 'rails_helper'
+
+RSpec.describe Users::OmniauthCallbacksController, type: :controller do
+
+  before { OmniAuth.config.test_mode = true }
+
+  before(:each) do
+    request.env['devise.mapping'] = Devise.mappings[:user]
+    request.env["omniauth.auth"] = OmniAuth.config.mock_auth[:facebook]
+  end
+
+  describe '#facebook' do
```

Comment:
No sure, but you should test such things here:
- If user with such email does not exist
  - check if user was created from `request.env['omniauth.auth']`
  - check if user was signed in
  - check redirect path
  - check if order was added to user
  - check flash message
- User with such email exists
  - check if user was signed in
  - check that new user wasn't created
  - check redirect path
  - check flash message

Also in which case user can't be persisted after `User.from_omniauth(request.env['omniauth.auth'])`?

Reply:
done, but the comment did not disappear